packages/orchestration/src/orchestration-api.ts

```diff
@@ -73,34 +72,34 @@ export type ChainAddress = {
  *
  * The methods available depend on the chain and its capabilities.
  */
-export type OrchestrationAccount<CI extends ChainInfo> = OrchestrationAccountI &
-  (CI extends CosmosChainInfo
-    ? CI['chainId'] extends `agoric${string}`
-      ? CosmosChainAccountMethods<CI> & LocalAccountMethods
-      : CosmosChainAccountMethods<CI>
-    : {});
+export type OrchestrationAccount<C extends keyof KnownChains> =
+  OrchestrationAccountI &
+    (KnownChains[C] extends CosmosChainInfo
+      ? KnownChains[C]['chainId'] extends `agoric${string}`
+        ? CosmosChainAccountMethods<KnownChains[C]> & LocalAccountMethods
+        : CosmosChainAccountMethods<KnownChains[C]>
+      : {});
 
 /**
  * An object for access the core functions of a remote chain.
  *
  * Note that "remote" can mean the local chain; it's just that
  * accounts are treated as remote/arms length for consistency.
  */
-export interface Chain<CI extends ChainInfo> {
-  getChainInfo: () => Promise<CI>;
+export interface Chain<C extends keyof KnownChains> {
+  getChainInfo: () => Promise<KnownChains[C]>;
 
   // "makeAccount" suggests an operation within a vat
   /**
    * Creates a new account on the remote chain.
    * @returns an object that controls a new remote account on Chain
    */
-  makeAccount: () => Promise<OrchestrationAccount<CI>>;
+  makeAccount: () => Promise<OrchestrationAccount<C>>;
   // FUTURE supply optional port object; also fetch port object
-
-  query: CI extends { icqEnabled: true }
-    ? ICQQueryFunction
-    : CI['chainId'] extends `agoric${string}`
-      ? QueryManyFn
+  query: C extends 'agoric'
+    ? QueryManyFn
+    : KnownChains[C] extends { icqEnabled: true }
+      ? ICQQueryFunction
       : never;
 
   // TODO provide a way to get the local denom/brand/whatever for this chain
```

Comment:
Can we delete these or create issues? They seem like noise at this point

```
  // "makeAccount" suggests an operation within a vat
  // FUTURE supply optional port object; also fetch port object
  // TODO provide a way to get the local denom/brand/whatever for this chain
```

Reply:
agree, issues or delete. Please figure out whether they're still relevant (if so file, if not delete)